src/bootchooser.c

```diff
@@ -1375,6 +1375,206 @@ static gboolean efi_get_state(RaucSlot* slot, gboolean *good, GError **error)
 	return TRUE;
 }
 
+/* Wrapper for get commands accessing custom script
+ *
+ * @param cmd What to input as command to the custom backend. Mandatory.
+   Valid values:
+   - get-primary
+   - get-state
+ * @param bootname slot.bootname
+ * @param ret_str Return string from stdout after running the command
+ * @param error Return location for a GError
+ */
+static gboolean custom_backend_get(const gchar *cmd, gchar *bootname, GString **ret_str, GError **error)
+{
+	g_autoptr(GSubprocess) sub = NULL;
+	GError *ierror = NULL;
+	g_autoptr(GBytes) stdout_buf = NULL;
+	gint ret;
+	gsize size;
+	const char *data;
+	gchar *backend_name = r_context()->config->custom_bootloader_backend;
+
+	g_return_val_if_fail(cmd, FALSE);
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+	if (bootname)
+		sub = g_subprocess_new(G_SUBPROCESS_FLAGS_STDOUT_PIPE, &ierror, backend_name, cmd, bootname, NULL);
+	else
+		sub = g_subprocess_new(G_SUBPROCESS_FLAGS_STDOUT_PIPE, &ierror, backend_name, cmd, NULL);
+
+	if (!sub) {
+		g_propagate_prefixed_error(
+				error,
+				ierror,
+				"Failed to start : %s", backend_name);
+		return FALSE;
+	}
+
+	if (!g_subprocess_communicate(sub, NULL, NULL, &stdout_buf, NULL, &ierror)) {
+		g_propagate_prefixed_error(
+				error,
+				ierror,
+				"Failed to run : %s", backend_name);
+		return FALSE;
+	}
+	data = g_bytes_get_data(stdout_buf, &size);
+	*ret_str = g_string_new(data);
```

Comment:
Eh, is data necessarily nul-terminated? Don't you need to use g_string_new_len() using the size you got from g_bytes_get_data?